src/rpc/util.cpp

```diff
@@ -523,3 +531,14 @@ std::string RPCArg::ToString(const bool oneline) const
     }
     assert(false);
 }
+
+std::pair<int64_t, int64_t> ParseRange(const UniValue& value)
+{
+    if (value.isNum()) {
+        return {0, value.get_int64()};
+    }
+    if (value.isArray() && value.size() == 2 && value[0].isNum() && value[1].isNum()) {
+        return {value[0].get_int64(), value[1].get_int64()};
+    }
+    throw JSONRPCError(RPC_INVALID_PARAMETER, "Range must be specified as int or [int, int]");
```

Comment:
You could move the start <= end and >0 check here too.

Reply:
I've moved the `start <= end` check there. This function may be useful in places where negative numbers are acceptable, so I'm leaving the negative check in the callers.